core/core.go

```diff
@@ -0,0 +1,91 @@
+package core
+
+import "github.com/findy-network/findy-agent/agent/storage/api"
+
+type DID interface {
+
+	//	Resolve() DIDDoc
+	// Validate() error
+
+	KID() string
+	String() string
+	SignKey() any
+	Storage() api.AgentStorage
```

Comment:
It seems conceptually a bit weird that DID has accessor to Storage 🤔 Maybe it would make more sense to be in Pipe-level.

Reply:
I strongly disagree with that as well, but let's have the discussion.